waku/common/sqlite/migrations.nim

```diff
@@ -0,0 +1,138 @@
+{.push raises: [].}
+
+import 
+  std/[strutils, sequtils, os, algorithm],
+  stew/results,
+  chronicles
+import
+  ../sqlite
+  
+
+logScope:
+  topics = "sqlite"
+
+
+## Migration scripts
+
+proc getMigrationScriptVersion(path: string): DatabaseResult[int64] =
+  let name = extractFilename(path)
+  let parts = name.split("_", 1)
+
+  try:
+    let version = parseInt(parts[0])
+    return ok(version)
+  except ValueError:
+    return err("failed to parse file version: " & name)
+
+proc isSqlScript(path: string): bool =
+  path.toLower().endsWith(".sql")
+
+
+proc listSqlScripts(path: string): DatabaseResult[seq[string]] =
+  var scripts = newSeq[string]()
+
+  try: 
+    for scriptPath in walkDirRec(path):
+      if isSqlScript(scriptPath):
+        scripts.add(scriptPath)
+      else:
+        debug "invalid migration script", file=scriptPath
+  except OSError:
+    return err("failed to list migration scripts: " & getCurrentExceptionMsg())
+
+  return ok(scripts)
```

Comment:
ditto

Reply:
I missed this one. IMO we should go with no return unless necessary. Solving it